src/evox/operators/gaussian_process/regression.py

```diff
@@ -0,0 +1,138 @@
+# https://docs.jaxgaussianprocesses.com/examples/regression/
+
+import jax
+import jax.numpy as jnp
+import jax.random as jr
+from jax import jit
+import optax as ox
+
+try:
+    import gpjax as gpx
+    from gpjax.objectives import ConjugateMLL
+    from gpjax.mean_functions import Zero
+    from gpjax.kernels import RBF
+    from gpjax.fit import fit
+except ImportError as e:
+    original_error_msg = str(e)
+
+    def gpx(*args, **kwargs):
+        raise ImportError(
+            f'gpjax requires gpjax, but got "{original_error_msg}" when importing'
+        )
+
+    def RBF(*args, **kwargs):
+        raise ImportError(
+            f'RBF requires gpjax, but got "{original_error_msg}" when importing'
+        )
+
+    def Zero(*args, **kwargs):
+        raise ImportError(
+            f'Zero requires gpjax, but got "{original_error_msg}" when importing'
+        )
+
+    def ConjugateMLL(*args, **kwargs):
+        raise ImportError(
+            f'ConjugateMLL requires gpjax, but got "{original_error_msg}" when importing'
+        )
+
+    def fit(*args, **kwargs):
+        raise ImportError(
+            f'fit requires gpjax, but got "{original_error_msg}" when importing'
+        )
+
+
+# jax.config.update('jax_enable_x64', True)
+
+
+class GPRegression:
+    """
+    Gaussian Process Regreesion model using JAX, Optax and GPJAX.
+
+    This model applies Gaussian Process (GP) for regression tasks. It uses an RBF kernel and
+    a Zero mean function by default. The class allows for fitting the model to data and making predictions.
+    The GP's prior and posterior are defined, and the model is fitted using the specified optimization method.
+
+    Attributes:
+        kernel: The kernel function for the GP (default is RBF).
+                The difference between the kernels are shown in the link: https://docs.jaxgaussianprocesses.com/examples/intro_to_kernels/.
+        mean_fun: The mean function for the GP (default is Zero).
+        key: JAX random key for stochastic operations.
+        prior: The prior distribution of the GP, which is chosen by the user according to the specific problems.
+        likelihood: The likelihood function for the GP.
+        object: The objective function (default is Conjugate Marginal Log Likelihood).
+        posterior: The posterior distribution of the GP after applying the likelihood.
+    """
+
+    def __init__(
+        self,
+        kernel=RBF(),
```

Comment:
In python `kernel=RBF()` will evaluate RBF() at import time, which is really bad.
use `kernel="RBF"`, and in __init__, do
```python
def __init__(
        self,
        kernel="RBF",
        ....
        ):
        if kernel == "RBF":
            self.kernel = RBF()
```